test/OpenTelemetry.Instrumentation.AspNetCore.Tests/RouteTests/RoutingTests.cs

```diff
@@ -0,0 +1,195 @@
+// <copyright file="RoutingTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#nullable enable
+
+using System.Diagnostics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+using RouteTests.TestApplication;
+using Xunit;
+using static OpenTelemetry.Internal.HttpSemanticConventionHelper;
+
+namespace RouteTests;
+
+public class RoutingTests : IClassFixture<RoutingTestFixture>
+{
+    private const string OldHttpStatusCode = "http.status_code";
+    private const string OldHttpMethod = "http.method";
+    private const string HttpStatusCode = "http.response.status_code";
+    private const string HttpMethod = "http.request.method";
+    private const string HttpRoute = "http.route";
+
+    private readonly RoutingTestFixture fixture;
+    private readonly List<Activity> exportedActivities = new();
+    private readonly List<Metric> exportedMetrics = new();
+
+    public RoutingTests(RoutingTestFixture fixture)
+    {
+        this.fixture = fixture;
+    }
+
+    public static IEnumerable<object[]> TestData => RoutingTestCases.GetTestCases();
+
+    [Theory]
+    [MemberData(nameof(TestData))]
+    public async Task TestHttpRoute(RoutingTestCases.TestCase testCase, bool useLegacyConventions)
```

Comment:
nit: We can keep this simple. Do not test the old semantic conventions as we are going to remove them soon.

Reply:
I considered that. Question is whether we'll be doing another release of the instrumentation _before_ removing them. We discussed a timeline awhile back, but I can't recall if we intend for our next release to remove the old conventions. In either case, probably doesn't hurt to just leave this in place. It's a minor thing to clean up.

Reply:
We would do another release before removing. I mentioned because I don't see much value in testing older ones now. I am good with cleaning up later.